src/components/new-request/search.vue

```diff
@@ -236,24 +266,33 @@ export default class Search extends Mixins(CommonMixin, NrAffiliationMixin, Sear
     return (Designations[this.getEntityTypeCd]?.end || false)
   }
 
+  get isSociety (): boolean {
+    return (this.isSocietyEnabled() && this.getEntityTypeCd === EntityTypes.SO)
+  }
+
   get showCompanyTypeRadioButtons (): boolean {
-    if (!this.entity_type_cd && !this.getConversionType && !this.isFederal) return false
-    if (this.isConversion) {
-      return (!!this.getSearchBusiness && !!this.getConversionType)
-    }
-    const isSociety = (this.isSocietyEnabled() && this.getEntityTypeCd === EntityTypes.SO)
-    // check if numbered is not allowed or society NR name is required
-    if (!this.isNumberedEntityType || isSociety) {
-      this.setSearchCompanyType(CompanyTypes.NAMED_COMPANY)
-      return false
-    }
-    return true
+    // *** TODO: add your logic in the template instead of the spaghetti below
+    // if (!this.getEntityTypeCd && !this.getConversionType && !this.isFederal) return false
+    // if (this.isConversion) {
+    //   return (!!this.getSearchBusiness && !!this.getConversionType)
+    // }
+    // // check if numbered is not allowed or society NR name is required
+    // if (!this.isNumberedEntityType || this.isSociety) {
+    //   this.setSearchCompanyType(CompanyTypes.NAMED_COMPANY)
+    //   return false
+    // }
+    return false
```

Comment:
Commented out (same as Karim did in his PR) as we should have the (much simpler) logic in the template instead.